packages/routing-utils/src/types.ts

```diff
@@ -21,7 +21,39 @@ export type HasField = Array<
     }
 >;
 
-export type Source = {
+type MiddlewareIndexed =
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware: number;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareAt?: never;
+    }
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware?: never;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareAt: string;
+    }
+  | {
+      /**
+       * A middleware index in the `middleware` key under the build result
+       */
+      middleware?: never;
+      /**
+       * A middleware key within the `output` key under the build result
+       */
+      middlewareAt?: never;
+    };
```

Comment:
It might be good to add tests for this to make sure it works as you expect with the 3 variations

Reply:
👍